R/appDependencies.R

```diff
@@ -61,9 +62,48 @@
 #' @export
 appDependencies <- function(appDir = getwd(), appFiles = NULL) {
   appFiles <- listDeploymentFiles(appDir, appFiles)
+  appMetadata <- appMetadata(appDir, appFiles = appFiles)
+  if (!needsR(appMetadata)) {
+    return(data.frame(
+      Package = character(),
+      Version = character(),
+      Source = character(),
+      Repository = character(),
+      stringsAsFactors = FALSE
+    ))
+  }
+
   bundleDir <- bundleAppDir(appDir, appFiles)
   on.exit(unlink(bundleDir, recursive = TRUE), add = TRUE)
 
-  deps <- snapshotRDependencies(bundleDir)
+  extraPackages <- inferRPackageDependencies(appMetadata)
+  deps <- snapshotRDependencies(bundleDir, extraPackages)
   deps[c("Package", "Version", "Source", "Repository")]
 }
+
+needsR <- function(appMetadata) {
+  if (appMetadata$appMode %in% c("static", "tensorflow-saved-model")) {
+    return(FALSE)
+  }
+
+  # All non-Quarto content currently uses R by default.
+  # Currently R is only supported by the "knitr" engine, not "jupyter" or
+  # "markdown"
+  is.null(appMetadata$quartoInfo) ||
+    "knitr" %in% appMetadata$quartoInfo[["engines"]]
+}
+
+inferRPackageDependencies <- function(appMetadata) {
+  deps <- switch(appMetadata$appMode,
+    "rmd-static" = c("rmarkdown", if (appMetadata$hasParameters) "shiny"),
+    "quarto-static" = "rmarkdown",
```

Comment:
Note to self: I was curious about whether or not we could make "quarto-static" need only "knitr" rather than "rmarkdown". Even though Quarto clams to use the "knitr" engine, there are references to the rmarkdown package in the Quarto source. I cannot easily tell if any of those references are needed other than for Shiny document execution with `rmarkdown::run`.

TL;DR: Keep `rmarkdown` as a dependency for `quarto-static` that uses the `knitr` engine.